assets/js/components/simulation/QuestionnaireSimulation.jsx

```diff
@@ -197,18 +197,20 @@ class QuestionnaireSimulation extends Component<Props, State> {
   handleATMessage = (message) => {
     const { projectId, questionnaireId, mode } = this.props
     const { simulation } = this.state
-    if (simulation) {
+    let reply
```

Comment:
The following sounds a little odd to me: `reply` can take its value from `message` or `message.body` depending on the context. The type of `message` depends on the context? Or the type of `reply` depends on the context?
We may want to add some sort of type check or at least a comment to clarify the expected types.

Reply:
You're right. I don't like it either. There is a `Message|string` type but it feels wrong. The backend doesn't care about the type: it only receives a message (the `message.body`).

Let's make `message.type` optional? So we'd have `{ body: "1", type: "at" }` and `{ body: "timeout" }`, oh and we can also have `{ body: "stop" }` to avoid logging the stop message :)